### PR TITLE
fix: seed system roles

### LINKED ISSUE #1061: Seed system roles on org creation and backfill orgs created since May 2026
The seed migration file inserts the 7 system roles for every org that existed the moment the migration ran, but for any orgs created afterward the roles do not get created. This should probably be part of the `create-org-and-user.ts` bootstrap file and a backfill migration for any orgs created since May 2026.
(I wrote this I am a human!)

Context: because of this, newer orgs have zero `public.roles` rows and their users have `role_id = NULL`. Today this is masked by fallbacks (static role defaults, static permission map, and lazy materialization on first role edit), so permissions still work. But it means role rows and `users.role_id` are unreliable, which blocks referencing roles by `role_id` (e.g. queue access by role, Groups, deletable custom roles).
(Claude wrote this)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by taobojlen**
## Context & Requests for Reviewers

Fixes #1061 

In https://github.com/roostorg/coop/pull/528, we started working on more granular permissions. Roles + permissions are now persisted in the database.

However, new orgs created since then did *not* get the new DB-backed roles and permissions. Instead, they rely on defaults in the code, and the permissions only get created on-demand when the defaults are edited. This means that there are really 2 different implementations of permissions!

We can clean this up by getting every org to use the DB-backed roles and permissions so there's just one way to do it. This PR:

- Adds a data migration that creates the default roles + permissions for all orgs
- Updates the new-org script so that new orgs are created with the default roles + permissions

Once this is merged, we no longer need the non-DB, code `isFallback` path for permissions.

## Tests

I tested this manually by creating an org on main, editing a default role, then checking out this branch and running the data migration, and verifying the permissions in the DB.

## (Optional) Rollout Plan

N/A

## Checklist

_Only check items that apply to this PR; leave the rest unchecked._

- [ ] ~~**If you changed anything user-facing** (i.e. user interface or APIs):~~
  Did you update the CHANGELOG.md and related docs?

- [ ] ~~**If you changed `server/models/**/{ContentTypeModel,ActionModel,RuleModel,PolicyModel}.ts`:**~~
  Did you update the corresponding history tables and their triggers?

- [ ] ~~**If you changed `db/src/scripts/**` and used `CREATE TABLE`, `ADD COLUMN`, or `ALTER COLUMN`:**~~
  Are as many columns marked `NOT NULL` as possible? If some columns can sometimes be null depending on other columns, are there `CHECK` constraints capturing those relationships, and are these also reflected using unions in the associated Kysely types?

- [ ] ~~**If you added a new signal in `server/services/signalsService/signals/**`:**~~
  Did you classify every error case as a permanent error (`SignalPermanentError`, no retry) or a normal error (retryable)? Any case where the signal can't determine a score should be a `SignalPermanentError`.

## Summary by CodeRabbit

* **New Features**
  * System roles and their default permissions are now automatically created for each organization.
  * Existing users and invitations are linked to their corresponding system roles.
  * New organizations receive role setup automatically during creation.
  * Role provisioning is safe to repeat and preserves existing role assignments.

### CODE REVIEW

File: db/src/scripts/api-server-pg/2026.08.25T15.56.24.seed_missing_system_roles.sql [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    AND r.key = u.role
+    AND r.is_system = true;
+
+UPDATE public.invite_user_tokens AS t
+SET role_id = r.id
+FROM public.roles AS r
+WHERE t.role_id IS NULL
+    AND r.org_id = t.org_id
+    AND r.key = t.role
+    AND r.is_system = true;
```
Comment (julietshen) [inline db/src/scripts/api-server-pg/2026.08.25T15.56.24.seed_missing_system_roles.sql:108]: Not blocking!

BUT i think these two backfill UPDATEs are the only part of your change that touches existing rows, and they aren't covered by tests right now. the test file uses `seedSystemRolesForOrg`, which doesn't include this logic. Could we add a test that seeds an org, inserts a user and an invite token with a legacy `role` and `role_id IS NULL`, then exercises the backfill and confirms both get linked to the matching system role?

**Review by julietshen — APPROVED** [review]
LGTM! Maybe we should add a little heads up somewhere about how renaming a key
requires a data migration updating all three columns at the same time or things break?

File: db/src/scripts/api-server-pg/2026.08.25T15.56.24.seed_missing_system_roles.sql (hunk 1 above)
Comment (juanmrad) [reply db/src/scripts/api-server-pg/2026.08.25T15.56.24.seed_missing_system_roles.sql:108]: Normally I would agree with this assessment, but in practice adding tests to migrations is mostly dead code later own. This is something better tested manually ensuring this migration works as expected and that is all.

**Review by juanmrad — APPROVED** [review]
Thank you for this. Totally scaped my mind 😓
